frontend/src/metabase/parameters/utils/date-formatting.ts

```diff
@@ -126,16 +126,24 @@ function parseDateRangeValue(value: string) {
   return { start: moment(start, true), end: moment(end, true) };
 }
 
+function formatSingleDate(date: Moment) {
+  if (date.hours() || date.minutes()) {
+    return date.format("MMMM D, YYYY hh:mm a");
```

Comment:
We have [localization settings](https://www.metabase.com/docs/latest/configuring-metabase/localization#dates-and-times) that should be taken into account here.

### Time style (12/24h)

When I use 24h format:

1. the popover won't have the AM/PM switch but the value in the input will be formatted with AM/PM ([image](https://github.com/user-attachments/assets/26e40f29-22e9-4ca7-bb68-fe149640dcad)) 
2. in the notebook editor, the time inputs and filter names will still use AM/PM in them ([image](https://github.com/user-attachments/assets/7d54e778-3c5e-4c1c-ae27-ea7fe3370de1))

### Abbreviate names of days and months

When I turn it on:

3. it has no effect in native editor
4. :heavy_check_mark: it works as expected in the notebook editor ([image](https://github.com/user-attachments/assets/70087aa1-e05d-45ef-8146-ac7f9aafce9b))


----

:x: Issue (1) is introduced here.

Issue (3) is already there but it would be great to fix it here.

----

(2) is actually (BE + FE), so I reported them separately:
- #49076
- #49077

I also found:
- #49075

Reply:
Resolved, see: https://github.com/metabase/metabase/pull/49046#issuecomment-2434953696